### ISSUE TITLE
Animations using createAnimation fail to sync when user joins late

### ISSUE THREAD

**Issue by norybiak** [microsoft/mixed-reality-extension-sdk#529]
The new animation system will fail to properly update join-in-progress users.

I found the culprit after some digging: [cacheAnimationCreation() in session.ts](https://github.com/microsoft/mixed-reality-extension-sdk/blob/409bdaf08e7ac0883e20db2ce281670d78245cb9/packages/sdk/src/internal/adapters/multipeer/session.ts#L420) called from ['object-spawned' in rules.ts](https://github.com/microsoft/mixed-reality-extension-sdk/blob/409bdaf08e7ac0883e20db2ce281670d78245cb9/packages/sdk/src/internal/adapters/multipeer/rules.ts#L756).

Looking at the network traffic logs, 'object-spawned' is received some time after the app handles the animation initialization. cacheAnimationCreation() sets anim.update to undefined.

Users that join late try to access this cached update during the [syncing stage](https://github.com/microsoft/mixed-reality-extension-sdk/blob/409bdaf08e7ac0883e20db2ce281670d78245cb9/packages/sdk/src/internal/adapters/multipeer/protocols/clientSync.ts#L233) but the sync payload doesn't get sent since anim.update is undefined.

**Comment by stevenvergenz** [comment]
@norybiak The `update` field is only supposed to contain changes from the default. If it's null, then all clients will get the initialization state, as defined in the `create-animation` message. Still, this does sound real, even if this isn't the culprit. I'll investigate.

**Comment by stevenvergenz** [comment]
The animation functional test syncs for late joiners successfully. Could you share more details of your use case where it fails?

**Comment by norybiak** [comment]
I should have been more specific to which part of the animation system this bug applies to. The syncing issue I'm seeing is when an animation is created using `actor.createAnimation()`. 

I've figured out my problem and it comes down to having to wait for the animation to initialize before trying to update it.

The following is my test that produces the unexpected syncing issue:
```javascript
        myActor.createAnimation('test', 
        {
            keyframes: 
            [{
                time: 0,
                value: { transform: { local: { position: { x: 1 } } } }
            }, {
                time: 10,
                value: { transform: { local: { position: { x: 10 } } } }
            }],
            events: [],
            wrapMode: MRE.AnimationWrapMode.Loop
        });

        myActor.enableAnimation("test");
```

The above will work for the authoritative user and any user that's already in the space while the MRE loads. It fails for any user that joins later because `myActor.createAnimation` will override the update set by `myActor.enableAnimation` (as described from my initial comment). 

Waiting for the animation to initialize solves the problem. 
```javascript
        let anim = await asset.createAnimation('test', 
        {
            keyframes: 
            [{
                time: 0,
                value: { transform: { local: { position: { x: 1 } } } }
            }, {
                time: 10,
                value: { transform: { local: { position: { x: 10 } } } }
            }],
            events: [],
            wrapMode: MRE.AnimationWrapMode.Loop
        });

        anim.play();

or 

        asset.createAnimation('test', 
        {
            keyframes: 
            [{
                time: 0,
                value: { transform: { local: { position: { x: 1 } } } }
            }, {
                time: 10,
                value: { transform: { local: { position: { x: 10 } } } }
            }],
            events: [],
            wrapMode: MRE.AnimationWrapMode.Loop
        }).then((anim) => anim.play());
```

I suppose this isn't a bug but rather a pattern that can easily be overlooked since it doesn't show any errors when immediately playing the animation. 

[Clock sync test](https://github.com/microsoft/mixed-reality-extension-sdk/blob/red/packages/functional-tests/src/tests/clock-sync-test.ts), hello-world, and solar-system need to be updated since they don't wait for the animation to fully initialize before playing.

**Comment by stevenvergenz** [comment]
This should be fixed in MRE v0.18. Update and try again.
